cumulus_api/main.py

```diff
@@ -33,7 +33,10 @@ def __init__(self, use_os_env: bool = True, config_path: str = None, token: str
         self.INVOKE_BASE_URL = self.config["INVOKE_BASE_URL"].rstrip('/')
         self.cumulus_token = CumulusToken(config=config)
         self.TOKEN = token if token else self.cumulus_token.get_token()
-        self.HEADERS = {'Authorization': f'Bearer {self.TOKEN}'}
+        self.HEADERS = {
+            'Authorization': f'Bearer {self.TOKEN}',
+            'Cumulus-API-Version': '2',
```

Comment:
This header value should probably be implemented at the individual method level since most don't currently require it and there is potential for this to be a different value for different methods in the future (assumption given that some endpoints are currently at "1" while others are at 2.

Reply:
Based on the current Cumulus 16 implementation, I would have to agree. I took a closer look at the middleware code that is being used for version checking, and it is written such that each endpoint could provide an arbitrary version number. I don't know if there is any intent to use it this way, but it is supported.

At a first look, it appears that some refactoring would be required in order to apply the version header on an individual method level, so I am leaving that at the instance level until we get some more feedback from Cumulus.

Reply:
Discussion with Jonathan from Cumulus team: https://eosdis.slack.com/archives/C76M787MZ/p1691176763321379?thread_ts=1691084537.427589&cid=C76M787MZ 
There shouldn't be an issue with including this header for endpoints which don't need them, so we will adapt this approach to avoid unnecessary updates on a per method basis.